transmute between types of different sizes

Not sure what changed but this just started coming up in one of my CI tests
```

Compiling stm32h7xx-hal v0.8.0
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32h7xx-hal-0.8.0/src/rng.rs:110:67
    |
110 |                         let bytes: [$type; BATCH_SIZE] = unsafe { mem::transmute(random_word) };
    |                                                                   ^^^^^^^^^^^^^^
...
126 | rng_core!(usize);
    | ----------------- in this macro invocation
    |
    = note: source type: `u32` (32 bits)
    = note: target type: `[usize; 0]` (0 bits)
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```
I have been messing with my code, but pretty sure my changes didn't cause this.

--- a/src/rng.rs
+++ b/src/rng.rs
@@ -122,5 +122,5 @@ macro_rules! rng_core {
 rng_core!(u32, u16, u8);
 
 // Test host may have > 32-bit types, which we don't consider.
-#[cfg(not(test))]
+#[cfg(target_pointer_width = "32")]
 rng_core!(usize);

